Model/Feed.php

```diff
@@ -54,166 +65,210 @@ public function __construct(
         );
     }
 
-
-    // Process the product feed and update the progress file, in page sizes of 20 (or other if overriden)
-    function ProductFeed($storeId, $progressFileName, $feedFile, $doNdjson, $pageSize = 20)
+    /**
+     * Process the product feed and update the progress file, in page sizes
+     * of 1 by default to ensure fits POST size
+     * @param $pageSize integer
+     */
+    function sendProducts($pageSize = 1)
     {
-        
+        if (! $this->isInitialised()) {
+            return false;
+        }
+
+        $this->start(self::FEED_TYPE_PRODUCT);
+
+        $this->logger->debug("PureClarity: In Feed->sendProducts()");
         $productExportModel = $this->coreProductExportFactory->create();
-        $productExportModel->init($storeId);
+        $productExportModel->init($this->storeId);
+        $this->logger->debug("PureClarity: Initialised ProductExport");
 
         $currentPage = 0;
         $pages = 0;
         $feedProducts = [];
-        $this->coreHelper->setProgressFile($progressFileName, 'product', 0, 1);
+        $this->coreHelper->setProgressFile($this->progressFileName, self::FEED_TYPE_PRODUCT, 0, 1);
+        $this->logger->debug("PureClarity: Set progress");
 
+        // loop through products, POSTing string for each page as it loops through
         $isFirst = true;
-        $count = 1;
         do {
             $result = $productExportModel->getFullProductFeed($pageSize, $currentPage);
+            $this->logger->debug("PureClarity: Got result from product export model");
+
             $pages = $result["Pages"];
         
-            $json = "";
+            $json = ($isFirst ? ',"Products":[' : "");
             foreach ($result["Products"] as $product) {
-                if ($isFirst == false && !$doNdjson) {
+                if (! $isFirst) {
                     $json .= ',';
                 }
-                $isFirst=false;
-                $json .= $this->coreHelper->formatFeed($product, 'json') . ($doNdjson?PHP_EOL:'');
+                $isFirst = false;
+                $json .= $this->coreHelper->formatFeed($product, 'json');
             }
-            fwrite($feedFile, $json);
+            if (($currentPage) >= $pages) {
+                $json .= ']';
```

Comment:
End of array if end of products - using >= rather than == just adds additional safeguard